### PR TITLE
Static assert methods

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | n/a
| License       | MIT
| Doc PR        | n/a

Based on the work for static data providers, (non-static data providers are deprecated since PHPUnit 10) I spotted, that we are using many method with `$this` instead of `self`.

I did some manual search/replace, to figure out if you are interested in such a PR @nicolas-grekas. If yes, I will try to find a rule and setup rector (https://github.com/rectorphp/rector/issues/7682)

I found the rector 👏 

Can be executed via:
```
tools/vendor/bin/rector process --config=tools/rector.php
```

### CODE REVIEW

File: src/Symfony/Component/Console/Tests/ApplicationTest.php [hunk 1]
```diff
@@ -509,12 +509,12 @@ public function testDontRunAlternativeNamespaceName()
...
-                                                          
-  Did you mean this?                                      
-      foo                                                 
-                                                          
+
+  There are no commands defined in the "foos" namespace.
+
+  Did you mean this?
+      foo
+
```
Comment (OskarStark) [inline src/Symfony/Component/Console/Tests/ApplicationTest.php]: Should be kept

**Comment by ro0NL** [comment]
There's already a CS fixer also, apparently it's not part of the symfony ruleset ;)

**Comment by OskarStark** [comment]
Yes, we can also fix this via PHP-Cs-Fixer, but we should rethink this rule, because if dataProviders should be static, you need to use static assertions anyway, for example like done in:
https://github.com/symfony/symfony/pull/48709/files#diff-48a3fd89675e4bfdd69f230c244e674e85794f5f1fecbef4e45cc940970b1b1c

**Comment by OskarStark** [comment]
Closing as not wanted by @nicolas-grekas

**Pull request closed without merging**